### ISSUE TITLE
Searching for specific hostgroup_name does not work

### ISSUE THREAD

**Issue by shakalandy** [Icinga/icingaweb2#2971]
## Expected Behavior
Using the search bar searching for a configured _hostgroup_name_ does not work, whereas searching for the _alias_ is working.
We often use shortnames in the _hostgroup_name_ variable and expect to find the hostgroup by searching for it.

## Current Behavior
searching after the configured _hostgroup_name_ nothing is found.

## Possible Solution
take the _hostgroup_name_ variable in account when searching.

## Steps to Reproduce (for bugs)
1. Configure a Hostgroup with _hostgroup_name_ = "all_hosts" and _alias" "all aws hosts"
2. Open icingaweb
3. search for "all_"
4. nothing found.

## Your Environment

* Icinga Web 2.4.1
* Icinga 1.14.0

**Comment by shakalandy** [comment]
This is for an example hostgroup with the name "gro_all"

```
QUERY
=====
SELECT statussummary.hostgroup_alias COLLATE latin1_general_ci AS
hostgroup_alias, statussummary.hostgroup_name, SUM(CASE WHEN object_type =
'host' AND state = 1 AND handled != 0 THEN 1 ELSE 0 END) AS
hosts_down_handled, SUM(CASE WHEN object_type = 'host' AND state = 1 AND
handled = 0 THEN 1 ELSE 0 END) AS hosts_down_unhandled, SUM(CASE WHEN
object_type = 'host' AND state = 99 THEN 1 ELSE 0 END) AS hosts_pending,
SUM(CASE WHEN object_type = 'host' THEN 1 ELSE 0 END) AS hosts_total,
SUM(CASE WHEN object_type = 'host' AND state = 2 AND handled != 0 THEN 1
ELSE 0 END) AS hosts_unreachable_handled, SUM(CASE WHEN object_type =
'host' AND state = 2 AND handled = 0 THEN 1 ELSE 0 END) AS
hosts_unreachable_unhandled, SUM(CASE WHEN object_type = 'host' AND state =
0 THEN 1 ELSE 0 END) AS hosts_up, SUM(CASE WHEN object_type = 'service' AND
state = 2 AND handled + host_state > 0 THEN 1 ELSE 0 END) AS
services_critical_handled, SUM(CASE WHEN object_type = 'service' AND state
= 2 AND handled + host_state = 0 THEN 1 ELSE 0 END) AS
services_critical_unhandled, SUM(CASE WHEN object_type = 'service' AND
state = 0 THEN 1 ELSE 0 END) AS services_ok, SUM(CASE WHEN object_type =
'service' AND state = 99 THEN 1 ELSE 0 END) AS services_pending, SUM(CASE
WHEN object_type = 'service' THEN 1 ELSE 0 END) AS services_total, SUM(CASE
WHEN object_type = 'service' AND state = 3 AND handled + host_state > 0
THEN 1 ELSE 0 END) AS services_unknown_handled, SUM(CASE WHEN object_type =
'service' AND state = 3 AND handled + host_state = 0 THEN 1 ELSE 0 END) AS
services_unknown_unhandled, SUM(CASE WHEN object_type = 'service' AND state
= 1 AND handled + host_state > 0 THEN 1 ELSE 0 END) AS
services_warning_handled, SUM(CASE WHEN object_type = 'service' AND state =
1 AND handled + host_state = 0 THEN 1 ELSE 0 END) AS
services_warning_unhandled FROM (SELECT CASE WHEN
(hs.problem_has_been_acknowledged + hs.scheduled_downtime_depth) > 0 THEN 1
ELSE 0 END AS handled, NULL AS host_state, hg.alias COLLATE
latin1_general_ci AS hostgroup_alias, hgo.name1 AS hostgroup_name, ('host')
AS object_type, CASE WHEN hs.current_state = 0 THEN CASE WHEN
hs.has_been_checked = 0 OR hs.has_been_checked IS NULL THEN 16 ELSE 0 END +
CASE WHEN hs.problem_has_been_acknowledged = 1 THEN 2 ELSE CASE WHEN
hs.scheduled_downtime_depth > 0 THEN 1 ELSE 4 END END ELSE CASE WHEN
hs.has_been_checked = 0 OR hs.has_been_checked IS NULL THEN 16 WHEN
hs.current_state = 1 THEN 32 WHEN hs.current_state = 2 THEN 64 ELSE 256 END
+ CASE WHEN hs.problem_has_been_acknowledged = 1 THEN 2 ELSE CASE WHEN
hs.scheduled_downtime_depth > 0 THEN 1 ELSE 4 END END END AS severity, CASE
WHEN hs.has_been_checked = 0 OR hs.has_been_checked IS NULL THEN 99 ELSE
hs.current_state END AS state, UNIX_TIMESTAMP(hs.last_state_change) AS
state_change FROM icinga_hosts AS h
 INNER JOIN icinga_objects AS ho ON ho.object_id = h.host_object_id AND
ho.is_active = 1 AND ho.objecttype_id = 1
 INNER JOIN icinga_hoststatus AS hs ON hs.host_object_id = ho.object_id
 LEFT JOIN icinga_hostgroup_members AS hgm ON hgm.host_object_id =
ho.object_id
 LEFT JOIN icinga_hostgroups AS hg ON hg.hostgroup_id = hgm.hostgroup_id
 LEFT JOIN icinga_objects AS hgo ON hgo.object_id = hg.hostgroup_object_id
AND hgo.is_active = 1 AND hgo.objecttype_id = 3 WHERE (hgo.name1 IS NOT
NULL) AND ( (hg.alias COLLATE latin1_general_ci LIKE '%gro_all%') ) GROUP
BY ho.object_id,
	h.host_id,
	hgo.object_id,
	hg.hostgroup_id UNION ALL SELECT CASE WHEN
(ss.problem_has_been_acknowledged + ss.scheduled_downtime_depth +
COALESCE(hs.current_state, 0)) > 0 THEN 1 ELSE 0 END AS handled, CASE WHEN
hs.has_been_checked = 0 OR hs.has_been_checked IS NULL THEN 99 ELSE CASE
WHEN hs.state_type = 1 THEN hs.current_state ELSE hs.last_hard_state END
END AS host_state, hg.alias COLLATE latin1_general_ci AS hostgroup_alias,
hgo.name1 AS hostgroup_name, ('service') AS object_type, NULL AS severity,
CASE WHEN ss.has_been_checked = 0 OR ss.has_been_checked IS NULL THEN 99
ELSE ss.current_state END AS state, UNIX_TIMESTAMP(ss.last_state_change) AS
state_change FROM icinga_objects AS so
 INNER JOIN icinga_services AS s ON s.service_object_id = so.object_id AND
so.is_active = 1 AND so.objecttype_id = 2
 INNER JOIN icinga_hoststatus AS hs ON hs.host_object_id = s.host_object_id
 INNER JOIN icinga_servicestatus AS ss ON ss.service_object_id =
so.object_id
 LEFT JOIN icinga_hostgroup_members AS hgm ON hgm.host_object_id =
s.host_object_id
 LEFT JOIN icinga_hostgroups AS hg ON hg.hostgroup_id = hgm.hostgroup_id
 LEFT JOIN icinga_objects AS hgo ON hgo.object_id = hg.hostgroup_object_id
AND hgo.is_active = 1 AND hgo.objecttype_id = 3 WHERE (hgo.name1 IS NOT
NULL) AND ( (hg.alias COLLATE latin1_general_ci LIKE '%gro_all%') ) GROUP
BY so.object_id,
	s.service_id,
	hgo.object_id,
	hg.hostgroup_id) AS statussummary GROUP BY statussummary.hostgroup_name,
	statussummary.hostgroup_alias ORDER BY hostgroup_alias COLLATE
latin1_general_ci ASC

COUNT
=====
SELECT COUNT(*) AS cnt FROM (SELECT statussummary.hostgroup_alias COLLATE
latin1_general_ci AS hostgroup_alias, statussummary.hostgroup_name,
SUM(CASE WHEN object_type = 'host' AND state = 1 AND handled != 0 THEN 1
ELSE 0 END) AS hosts_down_handled, SUM(CASE WHEN object_type = 'host' AND
state = 1 AND handled = 0 THEN 1 ELSE 0 END) AS hosts_down_unhandled,
SUM(CASE WHEN object_type = 'host' AND state = 99 THEN 1 ELSE 0 END) AS
hosts_pending, SUM(CASE WHEN object_type = 'host' THEN 1 ELSE 0 END) AS
hosts_total, SUM(CASE WHEN object_type = 'host' AND state = 2 AND handled
!= 0 THEN 1 ELSE 0 END) AS hosts_unreachable_handled, SUM(CASE WHEN
object_type = 'host' AND state = 2 AND handled = 0 THEN 1 ELSE 0 END) AS
hosts_unreachable_unhandled, SUM(CASE WHEN object_type = 'host' AND state =
0 THEN 1 ELSE 0 END) AS hosts_up, SUM(CASE WHEN object_type = 'service' AND
state = 2 AND handled + host_state > 0 THEN 1 ELSE 0 END) AS
services_critical_handled, SUM(CASE WHEN object_type = 'service' AND state
= 2 AND handled + host_state = 0 THEN 1 ELSE 0 END) AS
services_critical_unhandled, SUM(CASE WHEN object_type = 'service' AND
state = 0 THEN 1 ELSE 0 END) AS services_ok, SUM(CASE WHEN object_type =
'service' AND state = 99 THEN 1 ELSE 0 END) AS services_pending, SUM(CASE
WHEN object_type = 'service' THEN 1 ELSE 0 END) AS services_total, SUM(CASE
WHEN object_type = 'service' AND state = 3 AND handled + host_state > 0
THEN 1 ELSE 0 END) AS services_unknown_handled, SUM(CASE WHEN object_type =
'service' AND state = 3 AND handled + host_state = 0 THEN 1 ELSE 0 END) AS
services_unknown_unhandled, SUM(CASE WHEN object_type = 'service' AND state
= 1 AND handled + host_state > 0 THEN 1 ELSE 0 END) AS
services_warning_handled, SUM(CASE WHEN object_type = 'service' AND state =
1 AND handled + host_state = 0 THEN 1 ELSE 0 END) AS
services_warning_unhandled FROM (SELECT CASE WHEN
(hs.problem_has_been_acknowledged + hs.scheduled_downtime_depth) > 0 THEN 1
ELSE 0 END AS handled, NULL AS host_state, hg.alias COLLATE
latin1_general_ci AS hostgroup_alias, hgo.name1 AS hostgroup_name, ('host')
AS object_type, CASE WHEN hs.current_state = 0 THEN CASE WHEN
hs.has_been_checked = 0 OR hs.has_been_checked IS NULL THEN 16 ELSE 0 END +
CASE WHEN hs.problem_has_been_acknowledged = 1 THEN 2 ELSE CASE WHEN
hs.scheduled_downtime_depth > 0 THEN 1 ELSE 4 END END ELSE CASE WHEN
hs.has_been_checked = 0 OR hs.has_been_checked IS NULL THEN 16 WHEN
hs.current_state = 1 THEN 32 WHEN hs.current_state = 2 THEN 64 ELSE 256 END
+ CASE WHEN hs.problem_has_been_acknowledged = 1 THEN 2 ELSE CASE WHEN
hs.scheduled_downtime_depth > 0 THEN 1 ELSE 4 END END END AS severity, CASE
WHEN hs.has_been_checked = 0 OR hs.has_been_checked IS NULL THEN 99 ELSE
hs.current_state END AS state, UNIX_TIMESTAMP(hs.last_state_change) AS
state_change FROM icinga_hosts AS h
 INNER JOIN icinga_objects AS ho ON ho.object_id = h.host_object_id AND
ho.is_active = 1 AND ho.objecttype_id = 1
 INNER JOIN icinga_hoststatus AS hs ON hs.host_object_id = ho.object_id
 LEFT JOIN icinga_hostgroup_members AS hgm ON hgm.host_object_id =
ho.object_id
 LEFT JOIN icinga_hostgroups AS hg ON hg.hostgroup_id = hgm.hostgroup_id
 LEFT JOIN icinga_objects AS hgo ON hgo.object_id = hg.hostgroup_object_id
AND hgo.is_active = 1 AND hgo.objecttype_id = 3 WHERE (hgo.name1 IS NOT
NULL) AND ( (hg.alias COLLATE latin1_general_ci LIKE '%gro_all%') ) GROUP
BY ho.object_id,
	h.host_id,
	hgo.object_id,
	hg.hostgroup_id UNION ALL SELECT CASE WHEN
(ss.problem_has_been_acknowledged + ss.scheduled_downtime_depth +
COALESCE(hs.current_state, 0)) > 0 THEN 1 ELSE 0 END AS handled, CASE WHEN
hs.has_been_checked = 0 OR hs.has_been_checked IS NULL THEN 99 ELSE CASE
WHEN hs.state_type = 1 THEN hs.current_state ELSE hs.last_hard_state END
END AS host_state, hg.alias COLLATE latin1_general_ci AS hostgroup_alias,
hgo.name1 AS hostgroup_name, ('service') AS object_type, NULL AS severity,
CASE WHEN ss.has_been_checked = 0 OR ss.has_been_checked IS NULL THEN 99
ELSE ss.current_state END AS state, UNIX_TIMESTAMP(ss.last_state_change) AS
state_change FROM icinga_objects AS so
 INNER JOIN icinga_services AS s ON s.service_object_id = so.object_id AND
so.is_active = 1 AND so.objecttype_id = 2
 INNER JOIN icinga_hoststatus AS hs ON hs.host_object_id = s.host_object_id
 INNER JOIN icinga_servicestatus AS ss ON ss.service_object_id =
so.object_id
 LEFT JOIN icinga_hostgroup_members AS hgm ON hgm.host_object_id =
s.host_object_id
 LEFT JOIN icinga_hostgroups AS hg ON hg.hostgroup_id = hgm.hostgroup_id
 LEFT JOIN icinga_objects AS hgo ON hgo.object_id = hg.hostgroup_object_id
AND hgo.is_active = 1 AND hgo.objecttype_id = 3 WHERE (hgo.name1 IS NOT
NULL) AND ( (hg.alias COLLATE latin1_general_ci LIKE '%gro_all%') ) GROUP
BY so.object_id,
	s.service_id,
	hgo.object_id,
	hg.hostgroup_id) AS statussummary GROUP BY statussummary.hostgroup_name,
	statussummary.hostgroup_alias) AS t
```

**Comment by lippserd** [comment]
Thanks for the query. I thought that we had this feature for ages but actually only for hosts and services the object name is taken into account when searching. Recently we've introduced `hostgroup_name` and `servicegroup_name` as additional search columns via e42d86a4e185fc4f8d7e38e64f23705177ee1852. But I think it's best to relax this to their case insensitive versions. We'll fix this with version v2.5.0 which is scheduled for the end of November.
